start

```diff
@@ -542,10 +572,18 @@ function upgrade_local() {
 }
 
 function start_optional_services() {
-  # Wait for server
-  # TODO: Replace with checking if supervisord is up.
-  while ! echo "Stellar-core http server listening!" | nc localhost 11626 &> /dev/null; do sleep 1; done
+  while supervisorctl status > /dev/null; [ $? -eq 4 ]; do
+    sleep 1
```

Comment:
If this broke, nothing would be visible from this function specifically. But this is also in a background process and if supervisord didn't start up, which is in the foreground, the container would shut down immediately because it is pid 1, so I don't think we'd get much value out of a log line.